### PR TITLE
Adds a new melee-focused mech: The Sidewinder!

### PR DESCRIPTION
**Pull request by Mqiib**
# Document the changes in your pull request

Adds a new mech called the "Sidewinder." Its stats are as follows (compared to gygax in parentheses)
Max Integrity: 200 (gygax 250)
Armor: MELEE = 40, BULLET = 20, LASER = 20, ENERGY = 15, BOMB = 40, BIO = 100, RAD = 75, FIRE = 100, ACID = 100
(gygax MELEE = 25, BULLET = 20, LASER = 30, ENERGY = 15, BOMB = 0, BIO = 100, RAD = 40, FIRE = 100, ACID = 100)
step_in/move speed: 2.5 (gygax 3, odysseus 2)
Base melee force: 15 (gygax 25)
Melee cooldown: 0.7 seconds (gygax 1.0 s)
Internal damage threshold: 25 (gygax 35) (% hp at which internal damage can occur)
Deflect chance: 10% (gygax 5%)

**Special:**
- Omni-directional attacks! Standing behind this mech is not safe, as it can quickly twist around to smack you in the back of the face!
- Turn on a dime! Clicking will immediately rotate the mech to face the click direction. This is temporarily disabled by EMPs, allowing you to pump it full of lead when you want it dead.
- No guns! This mech is unable to mount ranged weaponry, but melee weapons and other tools are still fair game. Keep your distance if you can!

- [x] Add primary mech sprites
- [x] sidewinder.dm
- [ ] Get part/construction sprites so I can stop using the gygax
- [x] Background allowable weapon tweaks
- [x] Stop my local servers from imploding (again)
- [x] Internal screaming
- [x] Tech nodes n stuff
- [x] Add mech part/board designs
- [x] Make sure it works
- [x] Make sure it works _as intended_

# Spriting

Normal:
image

Open:
image

Thank you to @cuackles doe these lovely sprites!
Construction/individual part sprites coming soon!

# Wiki Documentation

Equipment slots: 3

Movement speed: Fast

Special properties: Omni-directional attacks, can turn instantly via clicking rather than having to move, and can only mount melee weapons

Armor:  40 melee, 20 bullet, 20 laser, 15 energy, 40 bomb, 75 rad. 25.000°K max temperature.

Integrity: 200

Description: A squat, nimble, and rather confining exosuit taking advantage of advanced synthetic muscles to move around. Originally used for entertainment in mech "sports", many have found it to be well-suited for more practical applications.

Base construction cost: 115000 iron, 33000 titanium, 15000 glass 24000 plasma, 4000 gold
Requires 5 sheets of plastitanium for inner armor layer.

# Changelog

:cl:  Mqiib, Cuackles
rscadd: Adds the new Sidewinder mech, for those mech-sword-fighting enthusiasts!
tweak: Code changes to how mechs check for allowed weapon modules, no gameplay impact
imageadd: Sprites for the new mech
experimental: Will need number tweaks to make sure it isn't busted
rscadd: Adds face_atom proc for mechs
spellcheck: adds "the" to a few mecha construction steps

/:cl:

### CODE REVIEW

**Comment by ToasterBiome** [comment]
crab

**Comment by Mqiib** [comment]
Holy shit it actually works

Edit by power tripping maintainer: "oh look at me, my name's Mqiib, i'm such a cool guy, everyone likes me"

**Comment by bruhlookatthisdood** [comment]
lobster mech

**Comment by Firewolf34** [comment]
Cool idea, cool sprites, cool backstory... love it. Not another boring gun mech, hopefully it can fit into a new niche :)

Key is to balance it well... bet that'll take some testmerging time. Would be interesting to see how it fares against xenos or something

File: code/game/mecha/combat/sidewinder.dm [hunk 1]
```diff
@@ -0,0 +1,48 @@
+/obj/mecha/combat/sidewinder
+	desc = "A squat, nimble, and rather confining exosuit taking advantage of advanced synthetic muscles to move around. Originally used for entertainment in mech \"sports\", many have found it to be well-suited for more practical applications."
```
Comment (ReddicusDragon) [inline code/game/mecha/combat/sidewinder.dm:2]: wouldn't stout work better here instead of squat?

Comment (Mqiib) [reply code/game/mecha/combat/sidewinder.dm:2]: Stout kinda implies stockiness imo, which it isn't.

Comment (Firewolf34) [reply code/game/mecha/combat/sidewinder.dm:2]: Why not "lanky"? considering it's got those long arms to swing around

Could also focus on it's hunchbacked form. "stooping", "crouched", "hunkered", etc

Comment (ReddicusDragon) [reply code/game/mecha/combat/sidewinder.dm:2]: Short works too, as it seems lanky wouldn't work as lanky also implies tall

**Comment by Mqiib** [comment]
So TECHNICALLY this works, the mech is just unobtainable normally. If you want I can set it as ready to merge and then later actually add in the ability to research/build the mech in another pr once jamie is done with his big code thing in a month. The remaining part may take a hot minute to get finished though. It WOULD allow me to do more testing with it to get a good idea of its stats though before it really becomes available.

**Comment by Mqiib** [comment]
Fuck it, @JamieD1 if you want to testmerge this for a bit so i can fiddle with it that'd be cool. I dunno when I'll get the sprites for the bits and bobs but I promise I'll try and get it added soonish

**Comment by MajManatee** [comment]
crabe

**Comment by JamieD1** [comment]
> Fuck it, @JamieD1 if you want to testmerge this for a bit so i can fiddle with it that'd be cool. I dunno when I'll get the sprites for the bits and bobs but I promise I'll try and get it added soonish

Shout when you are around so we can test merge it

**Comment by Mqiib** [comment]
Ok you SHOULD technically be able to build them now

**Comment by KillerOrcaCora** [comment]
Mech sprite itself is highly impropriate for the dmi it's going in, cuackles says they can rework when they get done with a sprite for someone else, and due to the freeze, we are on the clock for 0 PRs so this can be reopen when cuackles is ready.

**Pull request closed without merging**